src/components/SelectionList/Search/ReportListItem.tsx

```diff
@@ -62,11 +65,27 @@ function ReportListItem<TItem extends ListItem>({
 
         const isFromSelfDM = transactionItem.reportID === CONST.REPORT.UNREPORTED_REPORT_ID;
 
-        const reportID =
+        let reportID =
             (!transactionItem.isFromOneTransactionReport || isFromSelfDM) && transactionItem.transactionThreadReportID !== CONST.REPORT.UNREPORTED_REPORT_ID
                 ? transactionItem.transactionThreadReportID
                 : transactionItem.reportID;
 
+        // If we're trying to open a legacy transaction without a transaction thread, let's create the thread and navigate the user
```

Comment:
New expenses are created without transaction thread proactively so this is not only legacy transactions 
```suggestion
        // If we're trying to open a transaction without a transaction thread, let's create the thread and navigate the user
```